src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -99,15 +99,20 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
         raise MutuallyExclusiveArgumentError('usage error: --deployment-source-url <url> | --deployment-local-git')
 
     docker_registry_server_url = parse_docker_image_name(deployment_container_image_name)
-
+    
     client = web_client_factory(cmd.cli_ctx)
+    plan_info = None
     if is_valid_resource_id(plan):
         parse_result = parse_resource_id(plan)
         plan_info = client.app_service_plans.get(parse_result['resource_group'], parse_result['name'])
     else:
-        plan_info = client.app_service_plans.get(name=plan, resource_group_name=resource_group_name)
+        plans = list(client.app_service_plans.list(detailed=True))
+        for user_plan in plans: 
+            if user_plan.name.lower() == plan.lower():
+                parse_result = parse_resource_id(user_plan.id)
+                plan_info = client.app_service_plans.get(parse_result['resource_group'], parse_result['name'])
     if not plan_info:
-        raise ResourceNotFoundError("The plan '{}' doesn't exist in the resource group '{}".format(plan,
+        raise ResourceNotFoundError("The plan '{}' doesn't exist in the resource group '{}'".format(plan,
```

Comment:
If we're searching for plans beyond the RG that was passed to the command, then we should update this error message to say something like `plan {} not found`